src/scenes/Event/store.js

```diff
@@ -555,34 +562,26 @@ export default class EventStore {
     }
   }
 
-  // used by confirm tx modal
-  confirm = () => {
-    const actionToPerform = {
-      BETTING: this.bet,
-      RESULT_SETTING: this.setResult,
-      VOTING: this.vote,
-      FINALIZING: this.finalize,
-    }[this.oracle.phase];
-    if (!isUndefined(actionToPerform)) return actionToPerform();
-    console.error(`NO ACTION FOR PHASE ${this.oracle.phase}`); // eslint-disable-line
-  }
-
   bet = async () => {
     await this.app.tx.addBetTx(this.oracle.topicAddress, this.oracle.address, this.selectedOption.idx, this.amount);
   }
 
   setResult = async () => {
-    const { checkAllowance, currentAddress, isAllowanceEnough } = this.app.wallet;
+    const { isAllowanceEnough } = this.app.wallet;
     const { topicAddress } = this.oracle;
     const oracleAddress = this.oracle.address;
     const optionIdx = this.selectedOption.idx;
     const amountSatoshi = decimalToSatoshi(this.amount);
-    const allowance = await checkAllowance(currentAddress, topicAddress);
 
-    if (isAllowanceEnough(allowance, amountSatoshi)) {
-      await this.app.tx.addSetResultTx(undefined, topicAddress, oracleAddress, optionIdx, amountSatoshi);
-    } else {
+    if (this.allowance > 0 && !isAllowanceEnough(this.allowance, amountSatoshi)) {
```

Comment:
Previously we did not reset for setResult, but it worked fine. And so I'm a little confused, what was the reason to implement this reset?

My mind tells me it's a preparation for ERC20?
So even though QRC does not need to reset, we still let it behave like ERC20, am I wrong?

Reply:
FYI, this is a super edge case that shouldn't normally happen for most people, but it did for me while I was testing a lot.

On one of the events in the Set Result phase, I had previously had set a bad approve amount due to testing. Set Result is supposed to be 100 BOT, or 10000000000 Botoshi. While implementing a feature/testing I approved for 100 Botoshi on this event (this won't happen to anyone now since the code works).

But, that being said, without the RESET_APPROVE call, my approval for this event would be stuck at 100 Botoshi. So I would never be able to set the result on this event unless I reset the approve back to 0 first, then approve for the correct 10000000000 Botoshi.

Granted this was for a super edge case that should only happen for developers, smart contracts are open for anyone to interact with it. So say if a power user (person with technical Solidity knowledge) decided to call approve for this event outside of our dapp, when they used our dapp they would be stuck without a way to ever set the result on this event. I know this is prob a case that will never happen, but I did it more for myself to be able to reset the approve on the events that I already set prior wrong approve amounts for.

Reply:
wow, thank you